File: microservices/temperature-processing/src/main/java/com/algasensors/temperature/processing/infra/kafka/sensor/TemperatureMessage.java
package com.algasensors.temperature.processing.infra.kafka.sensor;

import com.algasensors.temperature.processing.api.model.SensorId;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;

@Getter
public class TemperatureMessage {

    @JsonProperty("sensorId")
    private SensorId sensorId;

    @JsonProperty("temperature")
    private double temperature;

    @JsonProperty("timestamp")
    private String timestamp;

    @Override
    public String toString() {
        return "TemperatureMessage{" +
                "sensorId='" + sensorId + '\'' +
                ", temperature=" + temperature +
                ", timestamp='" + timestamp + '\'' +
                '}';
    }
}

File: microservices/sensor/src/main/kotlin/com/algasensors/sensor/kafka/TemperatureMessage.kt
package com.algasensors.sensor.kafka

import com.fasterxml.jackson.annotation.JsonProperty

data class TemperatureMessage(
    @JsonProperty("sensorId") val sensorId: String,
    @JsonProperty("temperature") val temperature: Double,
    @JsonProperty(value = "unit") val unit: String,
    @JsonProperty("timestamp") val timestamp: String
)